packages/core/admin/admin/src/content-manager/hooks/useDragAndDrop.js

```diff
@@ -46,49 +46,112 @@ export const useDragAndDrop = (
+      const hoverBoundingRect = objectRef.current.getBoundingClientRect();
+      const hoverMiddleY = (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2;
+      const clientOffset = monitor.getClientOffset();
+      const hoverClientY = clientOffset.y - hoverBoundingRect.top;
 
-        // Dragging downwards
-        if (dragIndex < newInd && hoverClientY < hoverMiddleY) {
+      if (typeof index === 'number') {
+        if (dragIndex === newIndex) {
+          // Don't replace items with themselves
           return;
         }
 
-        // Dragging upwards
-        if (dragIndex > newInd && hoverClientY > hoverMiddleY) {
-          return;
+        if (dropSensitivity === 'regular') {
+          // Dragging downwards
+          if (dragIndex < newIndex && hoverClientY < hoverMiddleY) {
+            return;
+          }
+
+          // Dragging upwards
+          if (dragIndex > newIndex && hoverClientY > hoverMiddleY) {
+            return;
+          }
         }
-      }
 
-      // Time to actually perform the action
-      onMoveItem(newInd, dragIndex);
-      item.index = newInd;
+        // Time to actually perform the action
+        onMoveItem(newIndex, dragIndex);
+        item.index = newIndex;
+      } else {
+        // Using numbers as indices doesn't work for nested list items with path like [1, 1, 0]
+        if (dropSensitivity === 'regular') {
+          // Indices comparison to find item position in nested list
+          const minLength = Math.min(dragIndex.length, newIndex.length);
+          let areEqual = true;
+          let isLessThan = false;
+          let isGreaterThan = false;
+
+          for (let i = 0; i < minLength; i++) {
+            if (dragIndex[i] < newIndex[i]) {
+              isLessThan = true;
+              areEqual = false;
+              break;
+            } else if (dragIndex[i] > newIndex[i]) {
+              isGreaterThan = true;
+              areEqual = false;
+              break;
+            }
+          }
+
+          // Don't replace items with themselves
+          if (areEqual && dragIndex.length === newIndex.length) {
+            return;
+          }
+          // Dragging downwards
+          if (isLessThan && !isGreaterThan && hoverClientY < hoverMiddleY) {
+            return;
+          }
+
+          // Dragging upwards
+          if (isGreaterThan && !isLessThan && hoverClientY > hoverMiddleY) {
+            return;
+          }
+        }
+        onMoveItem(newIndex, dragIndex);
+        item.index = newIndex;
+      }
     },
   });
 
-  const [{ isDragging }, dragRef, dragPreviewRef] = useDrag({
+  const getDragDirection = (monitor) => {
+    if (monitor.isDragging() && !monitor.didDrop()) {
+      const deltaY = monitor.getInitialClientOffset().y - monitor.getClientOffset().y;
+
+      if (deltaY > 0) return 'upwards';
+
+      if (deltaY < 0) return 'downwards';
+
+      return 'no movement';
+    }
+
+    return 'no dragging';
+  };
```

Comment:
This doesn't seem very friendly, i think it really should be returning `null` if theres no direction – why do you need to know the difference between you're moving it but not enough and its not being moved? I'm also thinking realistically you're now depending on people spelling `upwards` and `downwards` correctly all the time – why not export the constants at least? Is just knowing the direction enough? what about if you want a higher tolerance? I'm wondering if you should actually just return `deltaY` and let other people figure it out? What do you think? 🤔